extensions/true-fantom/network/network-v001.js

```diff
@@ -0,0 +1,477 @@
+              {
+                text: '(7) redirected?',
+                value: '7'
+              },
+              {
+                text: '(8) url',
+                value: '8'
+              }
+            ]
+          },
+          response_type: {  
+            acceptReporters: true,
+            items: [
+              {
+                text: '(1) text',
+                value: '1'
+              },
+              {
+                text: '(2) json',
+                value: '2'
+              },
+              {
+                text: '(3) status ok?',
+                value: '3'
+              },
+              {
+                text: '(4) status',
+                value: '4'
+              },
+              {
+                text: '(5) status text',
+                value: '5'
+              },
+              {
+                text: '(6) type',
+                value: '6'
+              },
+              {
+                text: '(7) redirected?',
+                value: '7'
+              },
+              {
+                text: '(8) url',
+                value: '8'
+              },
+              {
+                text: '(9) body used?',
+                value: '9'
+              }
+            ]
+          },
+          default: { 
+            acceptReporters: true,
+            items: [
+              {
+                text: 'default',
+                value: 'default'
+              }
+            ]
+          }
+        }
+      }
+    }
+    
+    connected_to_internet_block() {
+      return navigator.onLine;
+    }
+    browser_block() {
+      let has = input => navigator.userAgent.includes(input);
+      if (has('Firefox')) return 'Firefox';
+      if (has('SamsungBrowser')) return 'Samsung Internet';
+      if ((has('Opera') || has('OPR')) && has('GX')) return 'Opera GX';
+      if (has('Opera') || has('OPR')) return 'Opera';
+      if (has('Trident')) return 'Internet Explorer';
+      if (has('Edge')) return 'Edge Legacy';
+      if (has('Edg')) return 'Edge Chromium';
+      if (has('YaBrowser') || has('YaSearchBrowser')) return 'Yandex';
+      if (has('Miui')) return 'Mi Browser';
+      if (has('UBrowser')) return 'UC Browser';
+      if (has('Chrome')) return 'Chromium';
+      if (has('Safari')) return 'Safari';
+      return '';
+    }
+    current_url_block() {
+      return document.URL || '';
+    }
+    network_type_block() {
+      let type = navigator.connection.type;
+      return type === undefined || type === 'none' || type === 'unknown' ? '' : type;
+    }
+    network_generation_block() {
+      return navigator.connection.effectiveType || '';
+    }
+    downlink_speed_block() {
+      return navigator.connection.downlink || '';
+    }
+    downlink_max_speed_block() {
+      return navigator.connection.downlinkMax || '';
+    }
+    network_rtt_block() {
+      return navigator.connection.rtt || '';
+    }
+    ping_block({SERVER}) {
+      return cached_ping_web_socket(String(SERVER));
```

Comment:
I question whether this needs to be here, but since it seems like you want this to be a personal extension more than anything, I won't stop you from including it

Reply:
Thanks for understanding)

Ps: It's not that I want to make an extension just for me. On the contrary, I would be glad if someone will use it. What I want is to achieve practicality and flexibility, but this cannot be fully achieved without a personal directory.